app/utils/themeUtils.tsx

```diff
@@ -22,13 +22,16 @@ export const ThemeContext = createContext<'dark' | 'light'>(getTheme());
 export const ThemeContextListener = () => {
   const dispatch = useAppDispatch();
 
+  const handleThemeChange = () => dispatch(setTheme(getTheme()));
   useEffect(() => {
-    const handleThemeChange = () => dispatch(setTheme(getTheme()));
-
```

Comment:
Is there any reason why you didn't run `handleThemeChange` inside this useEffect? I haven't checked into how many times this useEffect is re-rendered so idk if it's a good idea - but seeing that `handleThemeChange` relies on `dispatch`, it could be prudent to call it when it changes (in case it's not properly initialized on the first run)

Reply:
I made it it's own useEffect since I only figured it should run once on startup, but it might be safer to run it inside this useEffect? Not sure how often it rerenders either. I can investigate...

Reply:
Did you find out any thing interesting?(: